crates/sui-config/src/genesis.rs

```diff
@@ -816,10 +868,26 @@ impl Builder {
     }
 }
 
-fn get_genesis_context(epoch_data: &EpochData) -> TxContext {
+// Create a Genesis Txn Context to be used when generating genesis objects by hashing all of the
+// inputs into genesis ans using that as our "Txn Digest". This is done to ensure that coin objects
+// created between chains are unique
+fn create_genesis_context(
+    epoch_data: &EpochData,
+    genesis_chain_parameters: &GenesisChainParameters,
+    genesis_validators: &[GenesisValidatorMetadata],
+    token_distribution_schedule: &TokenDistributionSchedule,
+) -> TxContext {
+    let mut hasher = DefaultHash::default();
+    hasher.update(b"sui-genesis");
+    hasher.update(&bcs::to_bytes(genesis_chain_parameters).unwrap());
+    hasher.update(&bcs::to_bytes(genesis_validators).unwrap());
+    hasher.update(&bcs::to_bytes(token_distribution_schedule).unwrap());
+    let hash = hasher.finalize();
+    let genesis_transaction_digest = TransactionDigest::new(hash.into());
```

Comment:
@kchalkias Can you provide guidance on how best to do this hashing?

Reply:
Ah maybe it's not going to solve anything, just help on readability.

Reply:
@kchalkias The only other input not included in this hash is the sui/move framework module bytecode. What are you feelings or thoughts for or against including that in the hash as well?